boards: st: stm32 add the SD disk name to boards with sdmmc node

FRASTM opened the pull request:
Add a disk name to the SDMMC node, as done for all other SD card compatibles in tree.

Like the PR https://github.com/zephyrproject-rtos/zephyr/pull/88691/ did for the stm32h747 disco
 
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/89081

FRASTM committed:
boards: st: stm32 add the SD disk name to boards with sdmmc node

Add a disk name to the SDMMC node, as done for all other SD card
compatibles in tree.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

erwango reviewed (changes requested):
LGTM, but if this prop is now mandatory, let's make it `required: true` in `dts/bindings/mmc/st,stm32-sdmmc.yaml`

JordanYates commented:
Sorry for the failures, weird that CI didn't pick it up on my PR. Change looks fine, I agree with @erwango RE making the property required

FRASTM committed:
dts: bindings: mmc stm32 sdmmc has a mandatory disk-name

This PR makes the property disk-name required
for the st,stm32-sdmmc compatible

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
with `disk-name` property "required"

FRASTM commented:
with more non-ST boards

erwango commented:
@de-nordic Do you want to remain as assignee ? Otherwise I can take it

erwango commented:
Taking assignment as this issue is blocking others in CI
